src/v/cloud_storage/tests/cache_test_fixture.h

```diff
@@ -26,26 +27,35 @@
 using namespace cloud_storage;
 using namespace std::chrono_literals;
 
+static inline std::filesystem::path get_cache_dir(std::filesystem::path p) {
+    return p / "test_cache_dir";
+}
+
 class cache_test_fixture {
 public:
     const std::filesystem::path KEY{"abc001/test_topic/test_cache_file.txt"};
     const std::filesystem::path KEY2{"abc002/test_topic2/test_cache_file2.txt"};
     const std::filesystem::path TEMP_KEY{
       "abc002/test_topic2/test_cache_file2.txt_0_0.part"};
     const std::filesystem::path WRONG_KEY{"abc001/test_topic/wrong_key.txt"};
-    const std::filesystem::path CACHE_DIR{"test_cache_dir"};
 
+    temporary_dir test_dir;
     cloud_storage::cache cache_service;
+    const std::filesystem::path CACHE_DIR;
 
     cache_test_fixture()
-      : cache_service(
-        CACHE_DIR, 1_MiB + 500_KiB, ss::lowres_clock::duration(1s)) {
+      : test_dir("test_cache_dir")
```

Comment:
nit: should probably include a random component

Reply:
it's added inside the temporary_dir c-tor